src/InsertCapableInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Dhii\Storage\Resource;
+
+use Psr\Container\ContainerInterface;
+use Traversable;
+
+/**
+ * Something that can insert records into storage.
+ *
+ * @since [*next-version*]
+ */
+interface InsertCapableInterface
+{
+    /**
+     * Inserts records into storage.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $records An array of record data sets, each mapping field name to value.
```

Comment:
Is there a reason why this is not `Traversable`?

Reply:
Can't remember the reason for changing this to just `array`.

Insertion RMs will most likely traverse this list. So I think `Traversable` should be supported. That said, the items of the list/traversable ought to be containers IMO.

What do you think? Does it make sense to you to change to `array[]|ArrayAccess[]|stdClass[]|Container[]|Traversable`?

Reply:
Yeap. Especially if you've already updated the implementation to use the new container util methods.

Reply:
Don't think so. But if not, I will.

Reply:
Just realized, if the implementation is going to be using the container helper methods, it can't really use `Traversable`, can it?